src/main/java/com/rackspace/salus/monitor_management/services/MonitorManagement.java

```diff
@@ -2101,23 +2106,50 @@ public void handleExpiredEnvoy(@Nullable String zoneTenantId, String zoneName, S
     handleNewEnvoyInZone(zoneTenantId, zoneName);
   }
 
+  /**
+   * Returns the list of ZoneAssignmentCount for a particular zone of a tenant.
+   *
+   * @param zoneTenantId
+   * @param zoneName
+   * @return
+   */
   public CompletableFuture<List<ZoneAssignmentCount>> getZoneAssignmentCounts(
       @Nullable String zoneTenantId, String zoneName) {
 
     final ResolvedZone zone = resolveZone(zoneTenantId, zoneName);
+    ZoneAllocationResolver zoneAllocationResolver = zoneAllocationResolverFactory.create();
 
-    return zoneAllocationResolverFactory.create()
-        .getZoneBindingCounts(zone)
-        .thenApply(bindingCounts ->
-            bindingCounts.entrySet().stream()
-                .map(entry ->
-                    new ZoneAssignmentCount()
-                        .setResourceId(entry.getKey().getResourceId())
-                        .setEnvoyId(entry.getKey().getEnvoyId())
-                        .setAssignments(entry.getValue()))
-                .collect(Collectors.toList()));
+    return zoneAllocationResolver.getActiveZoneBindingCounts(zone)
+        .thenApply(activePollerMap -> mapToZoneAssignmentCount(activePollerMap, true))
+        .thenCompose(assignmentCounts -> zoneAllocationResolver.getExpiringZoneBindingCounts(zone)
+            .thenApply(envoyResourcePairIntegerMap -> {
+              assignmentCounts
+                  .addAll(mapToZoneAssignmentCount(envoyResourcePairIntegerMap, false));
+              return assignmentCounts;
+            }));
   }
 
+  /**
+   * This method is used to convert the map of EnvoyResourcePair count to list of
+   * ZoneAssignmentCount.
+   *
+   * @param bindingCounts
+   * @param isConnected
+   * @return
+   */
+  public List<ZoneAssignmentCount> mapToZoneAssignmentCount(
```

Comment:
One final thing!  Looks like this method could be package private?  In addition to the updated javadoc I think that would help show this is more of a helper method than something that is called directly by other services.

```suggestion
  List<ZoneAssignmentCount> mapToZoneAssignmentCount(
```

Reply:
Ohh yes, Sorry to miss out on that.